src/main/scala/arcana/world/Repl.scala

```diff
@@ -0,0 +1,47 @@
+package arcana.world
+
+import scala.io.StdIn
+
+object Repl {
+  val version = "0.0"
+  case class ArgumentFlags(typeInfo: Boolean)
+
+  val fooEvaluat = (input: String) => println(input)
+  val fooTypeEvaluat = (input: String) => println(input + ": type")
+  def evaluatArguments(args: Array[String]): Unit = {
+    var typeInfo = false
+    for (arg <- args) {
+      arg match {
+        // Add any argument here
+        case "--type" => typeInfo = true
+      }
+    }
```

Comment:
```suggestion
    val needTypeInfo = args.foldLeft(false)((flag, arg) => flag || arg.equals("--type"))
```
Try to use `fold`, `reduce` and so on in the code.